### ISSUE TITLE
query

### ISSUE THREAD

**Issue by vira12x** [rushandgg/LSTM_Stacked_Autoencoder#1]
Hii, rushandgg, Is it an unsupervised method ? Why you are averaging at the last part of the code can u please let me know?
Thanks.

**Comment by rushandgg** [comment]
Hi vira12x, It's an unsupervised method.

As you can see from the code, Input size is 20, so output is also 20. 
I was thinking about how to handle this part.

It's not in the code, I tried to create a time seires by connecting the last values of the windows.
But when I made it like this, there was a lag compared to the actual value.

I want to figure out how far the current value is from the reference value. not lagging value.
That's why I got the average of 20 windows to pick a value compareable to the present.

This is just my idea, and if there's a better way, let me know.
Thanks!

**Comment by vira12x** [comment]
Thanks @rushandgg for your suggestions.
I have a continuous timeseries data, I want to denoise it using your code. If it is unsupervised I think model.fit should be 
history = model.fit(x_train, x_train, epochs=500, batch_size=10,validation_split=0.1,shuffle=True, verbose=1)
instead of
history = model.fit(x_train, y_train, epochs=500, batch_size=10,validation_split=0.1,shuffle=True, verbose=1)
if possible please provide your mail id. Thanks.

**Comment by rushandgg** [comment]
You are right. But you'll know when you run the code, X_train and y_train are the same. 
My mail address is bso9419@naver.com
Thanks.
